src/regmod/parameter.py

```diff
@@ -22,72 +21,29 @@ class Parameter:
 
     Parameters
     ----------
-    name : str
+    name
         Name of the parameter.
-    variables : List[Variable]
+    variables
         A list of variables that parametrized the parameter.
-    inv_link : Union[str, SmoothFunction]
+    inv_link
         Inverse link funcion to link variables to parameter.
-    use_offset : bool, optional
-        If `True` when build parameter, it will use the offset in the data
-        object. Default to `False`.
-    linear_gpriors : List[LinearGaussianPrior], optional
+    offset
+        If `offset=None` parameter will not use offset, when it is a string
+        it will look for the corresponding column in the data frame as the
+        offset.
+    linear_gpriors
         A list of linear Gaussian priors. Default to an empty list.
-    linear_upriors : List[LinearUniformPrior], optional
+    linear_upriors
         A list of linear Uniform priors. Default to an empty list.
 
-    Attributes
-    ----------
-    name : str
-        Name of the parameter.
-    variables : List[Variable]
-        A list of variables that parametrized the parameter.
-    inv_link : Union[str, SmoothFunction]
-        Inverse link funcion to link variables to parameter.
-    use_offset : bool
-        If `True` when build parameter, it will use the offset in the data
-        object.
-    linear_gpriors : List[LinearGaussianPrior]
-        A list of linear Gaussian priors.
-    linear_upriors : List[LinearUniformPrior]
-        A list of linear Uniform priors.
-    size : int
-        Size of the parameter. It equals to sum of the sizes of the variables.
-
-    Methods
-    -------
-    check_data(data)
-        Attach data to all variables.
-    get_mat(data)
-        Get the design matrix.
-    get_uvec()
-        Get the direct Uniform prior.
-    get_gvec()
-        Get the direct Gaussian prior.
-    get_linear_uvec()
-        Get the linear Uniform prior vector.
-    get_linear_gvec()
-        Get the linear Gaussian prior vector.
-    get_linear_umat()
-        Get the linear Uniform prior matrix.
-    get_linear_gmat()
-        Get the linear Gaussian prior matrix.
-    get_lin_param(coefs, data, mat=None, return_mat=False)
-        Get the parameter before apply the link function.
-    get_param(coefs, data, mat=None)
-        Get the parameter.
-    get_dparam(coefs, data, mat=None)
-        Get the derivative of the parameter.
-    get_d2param(coefs, data, mat=None)
-        Get the second order derivative of the parameter.
     """
 
     name: str
-    variables: List[Variable] = field(repr=False)
-    inv_link: Union[str, SmoothFunction] = field(repr=False)
-    use_offset: bool = field(default=False, repr=False)
-    linear_gpriors: List[LinearGaussianPrior] = field(default_factory=list, repr=False)
-    linear_upriors: List[LinearUniformPrior] = field(default_factory=list, repr=False)
+    variables: list[Variable] = field(default_factory=list, repr=False)
```

Comment:
Is using `list` instead of `List` backwards compatible?

Reply:
Ah, unfortunately it is not, this is a new feature in python 3.9
https://peps.python.org/pep-0585/
which I think we can assume all our package moving forward support `python>=3.9`